src/plugins/vis_type_timeseries/common/last_value_utils.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isArray, last, isEqual } from 'lodash';
+
+export const EMPTY_VALUE = [];
+export const DISPLAY_EMPTY_VALUE = '-';
+
+type ExtractValueFn<T> = (data: T[] | void) => T | null;
+const extractValue: ExtractValueFn<unknown> = (data) => (data && data[1]) ?? null;
+
+type GetLastValueFn<T> = (data: T | T[] | T[][] | void) => T | null;
+export const getLastValue: GetLastValueFn<unknown> = (data) => {
+  if (!isArray(data)) {
+    return data;
+  }
+
+  return extractValue(last(data));
+};
+
+type GetLastValueOrEmptyFn<T, U> = (data: T[][] | void) => T | U;
+export const getLastValueOrEmpty: GetLastValueOrEmptyFn<unknown, []> = (data) =>
+  getLastValue(data) ?? EMPTY_VALUE;
```

Comment:
this part is not entirely clear to me. I can see that `getLastValue(data)` is returning a simple value. Why EMPTY_VALUE is array?

Reply:
I needed some value, which would be clearly explaining, that data was empty and need to handle in a different way, than usual.  Sure, I can change it to null.